blogposts/2020/optimizing-a-code-intelligence-indexer.md

```diff
@@ -0,0 +1,179 @@
+---
+title: "Optimizing a code intelligence indexer"
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2020-08-17T08:00-07:00
+tags: [blog]
+slug: optimizing-a-code-intel-indexer
+heroImage: https://sourcegraphstatic.com/blog/lsif-go/lsif-go-improvements.png
+published: true
+---
+
+Sourcegraph is used by companies all over the repository number and size spectrum. Some companies employ a microservice approach. For us, that's easy mode. Small repositories are easy and generally fast _enough_ to index even with an LSIF indexer that performs poorly as code size grows. Other companies employ a monorepo approach. This is where the challenge lies for us - but we've made great strides with the recent [1.0 release](https://github.com/sourcegraph/lsif-go/releases/tag/v1.0.0) of [lsif-go](https://github.com/sourcegraph/lsif-go).
```

Comment:
It's not clear to me that "microservice" means "small repos". You can have a monorepo with microservices.

Instead of "employ a (microservice/monorepo) approach", say "have many smaller repositories" vs "have one big monorepo" for simplicity.

Be precise about what the "challenge" is: it's a performance challenge.

Reply:
Updated.